### ISSUE TITLE
K_SHORTEST_PATHS returns wrong results (Path is not between start and target vertices)

### ISSUE THREAD

**Issue by agorobtsov** [arangodb/arangodb#10949]
## My Environment

* __ArangoDB Version__:        3.6.0
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server 
* __Deployment Strategy__:   Manual Start
* __Configuration__:               
* __Infrastructure__:                own
* __Operating System__:        Windows 10
* __Total RAM in your machine__:  16Gb.
* __Disks in use__:      SSD
* __Used Package__:   other

## Component, Query & Data

__Affected feature__:
K_SHORTEST_PATH via AQL

__AQL query (if applicable)__:
```
FOR p IN ANY K_SHORTEST_PATHS 'network_elements/1371235' TO 'network_elements/1391622'
    links
    OPTIONS {
        weightAttribute: 'weight1',
        defaultWeight: 10000
    }
    LIMIT 1
RETURN { vertices: p.vertices[*]._id, edges: p.edges[*]._id, edges: p.edges[*]._from, edges: p.edges[*]._to, weight: p.weight}
```

__AQL explain (if applicable)__:
Query String (327 chars, cacheable: true):
```
 FOR p IN ANY K_SHORTEST_PATHS 'network_elements/1371235' TO 'network_elements/1391622'
     links
     OPTIONS {
         weightAttribute: 'weight1',
         defaultWeight: 10000
     }
     LIMIT 1
 RETURN { vertices: p.vertices[*]._id, edges: p.edges[*]._id, edges: p.edges[*]._from, edges: 
 p.edges[*]._to, weight: p.weight}

Execution plan:
 Id   NodeType                      Est.   Comment
  1   SingletonNode                    1   * ROOT
  2   KShortestPathsNode   4902227890625     - FOR p  /* path */ IN ANY K_SHORTEST_PATHS 'network_elements/1371235' /* startnode */ TO 'network_elements/1391622' /* targetnode */ links
  3   LimitNode                        1     - LIMIT 0, 1
  4   CalculationNode                  1     - LET #11 = { "vertices" : p.`vertices`[*].`_id`, "edges" : p.`edges`[*].`_id`, "edges" : p.`edges`[*].`_from`, "edges" : p.`edges`[*].`_to`, "weight" : p.`weight` }   /* simple expression */
  5   ReturnNode                       1     - RETURN #11

Indexes used:
 none

k shortest paths on graphs:
 Id   Vertex collections   Edge collections
  2                        links              

Optimization rules applied:
 none
```

__Dataset__:
exported via arangoexport:
vertices:
[network_elements.zip](https://github.com/arangodb/arangodb/files/4091961/network_elements.zip)
edges:
[links.zip](https://github.com/arangodb/arangodb/files/4091980/links.zip)

__Size of your Dataset on disk__:

__Replication Factor & Number of Shards (Cluster only)__:

## Steps to reproduce

1. 
2. 
3. 

__Problem__:
Returned result is 

["network_elements/1365942","network_elements/1365962","network_elements/1365873","network_elements/1365900","network_elements/1366042","network_elements/1365861","network_elements/1366049","network_elements/1365911","network_elements/1365970","network_elements/1365867","network_elements/1366022","network_elements/1366040","network_elements/1365903","network_elements/1366006","network_elements/1391621"] | ["links/2790395","links/2787906","links/2789303","links/2789865","links/2789588","links/2787763","links/2789549","links/2790374","links/2789615","links/2789419","links/2790255","links/2788379","links/2787834","links/3538207"] | 1
-- | -- | --

__Expected result__:

Path should be between 'network_elements/1371235' AND 'network_elements/1391622'

**Comment by maxkernbach** [comment]
Hi @agorobtsov,

Version 3.6.1, which includes a fix for the reported bug, has just been released.
In case the issue persists, please comment and we will reopen this ticket.

You can find the latest release in the [download section](https://www.arangodb.com/download/).

**Comment by agorobtsov** [comment]
Hi @maxkernbach 
Еhank you very much for the quick solution

**Comment by markuspf** [comment]
Note that the above comment is inaccurate. The fix for this issue did *not* appear in `3.6.1`, but will appear in `3.6.2` and `3.7`

**Comment by maxkernbach** [comment]
Hi @agorobtsov,

Sorry for the confusion about the bug-fix version.
ArangoDB version 3.6.2, which includes the fix, has just been released.
